src/storage/mvcc/reader.rs

```diff
@@ -295,22 +295,38 @@ impl<'a> MvccReader<'a> {
         }
     }
 
-    pub fn scan_lock<F>(&mut self, filter: F) -> Result<Vec<(Key, Lock)>>
+    #[allow(type_complexity)]
+    pub fn scan_lock<F>(&mut self,
+                        start: Option<Key>,
+                        filter: F,
+                        limit: Option<usize>)
+                        -> Result<(Vec<(Key, Lock)>, Option<Key>)>
         where F: Fn(&Lock) -> bool
     {
         try!(self.create_lock_cursor());
         let mut cursor = self.lock_cursor.as_mut().unwrap();
-        cursor.seek_to_first();
+        let ok = match start {
+            Some(ref x) => try!(cursor.seek(x)),
```

Comment:
https://github.com/pingcap/tikv/pull/1271/files#diff-a581b51c6f8b34d75b3ce3e7c8037894R494
This test uses different count of locks to test ResolveLock, different values of `start` should be covered.